### PR TITLE
Refactor extraction pipeline.2

### PR DESCRIPTION
**Pull request by redur**
Here the discussed proposition.

I think we can extend the thought even further and include the draw_predictions earlier. That would require that we load the ground truth earlier as well. I keep this for the future as the current implementation currently already serves for discussion.

### CODE REVIEW

File: src/stratigraphy/main.py [hunk 1]
```diff
@@ -110,8 +111,34 @@ def start_pipeline(
...
+                        )
+
+                        predictions[filename][f"page_{page_number}"] = {
+                            "layers": layer_predictions,
+                            "depths_materials_column_pairs": depths_materials_column_pairs,
+                        }
+
+                        if draw_lines:
+                            logger.info("Drawing lines on pdf pages.")
+                            draw_lines_on_pdfs(filename, page, geometric_lines)
```
Comment (redur) [inline src/stratigraphy/main.py:140]: maybe rename to: `draw_lines_on_page`

File: src/stratigraphy/line_detection.py [hunk 2]
```diff
@@ -88,26 +87,19 @@ def extract_lines(page: fitz.Page, line_detection_params: dict) -> list[Line]:
     return lines
 
 
-def draw_lines_on_pdfs(input_directory: Path, line_detection_params: dict):
+def draw_lines_on_pdfs(filename: str, page: fitz.Page, geometric_lines: list[Line]):
     """Draw lines on pdf pages and stores them as artifacts in mlflow.
 
+    Note: now the function draw_lines_on_pdfs may not even be needed any more.
```
Comment (redur) [inline src/stratigraphy/line_detection.py:93]: Not sure it's worth having a separate function that essentially only has two lines of code. The MLFLow warning can be thrown somewhere else as well.

**Comment by redur** [comment]
Now single files work as well without the temporary directory.

**Review by stijnvermeeren-swisstopo — CHANGES_REQUESTED** [review]

**Comment by redur** [comment]
Thank you. Comments are addressed.

**Review by stijnvermeeren-swisstopo — APPROVED** [review]

**Pull request merged**